### ISSUE TITLE
[VOXEL REMOVAL PREVIEW BUILD ROUND#7] More bugfixes

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#3771]
ROUND#7 - More Bugfixes

More bugfixes following your test!

TO DO before the release:
- ship HP and new destroy mechanism.

Some more bugfixes: 
- Build placement of Blast Door and Trap Door fixed. 
- window tetra symmetry fixed
- Blueprint are converting back to normal
- Beam issue fixed.
- Invisible tile issue fixed.
- Windows frame issue with new light tech
- Player issue with new light tech

**Comment by tsunamayo** [comment]
Build is u! It is taking forever to finish but we will get there...

**Comment by ProPeach** [comment]
Thanks for the speedy fixes! That regression one was particularly nasty, but we're getting close to stable enough for experimental 🎉

**Comment by Kaiser-Indrasil** [comment]
Alright, out with the old bugs, in with the new ones!

All the decals are invisible in the Codex:

![image](https://user-images.githubusercontent.com/63756057/117592382-31623b80-b130-11eb-9c8f-5f40402fa242.png)

The MMB radial menu shows a white paint, no matter what colour I'm facing:

![image](https://user-images.githubusercontent.com/63756057/117592436-68385180-b130-11eb-9044-f302d3bf398e.png)

Running doesn't actually accelerate the player's movement.

The moving part of Hinges doesn't spawn after the world has been reloaded. The child entity doesn't disappear, just the moving part of the Hinge:

![image](https://user-images.githubusercontent.com/63756057/117594160-645afe00-b135-11eb-82d7-da798e8c81d5.png)

The painting SFX for the coatings appears way off the target:

![image](https://user-images.githubusercontent.com/63756057/117593968-04645780-b135-11eb-9d2b-269e3bc0a90e.png)

The stretched light block casts light onto its surroundings only from one location. I suspect it's the inherent limitation of the lighting system though:

![image](https://user-images.githubusercontent.com/63756057/117594910-1e9f3500-b137-11eb-8c87-fd54ebc81581.png)

And aside from the old bugs like bad inter-entity hitboxes and volumetric sensors that don't turn off, that's all the bugs!

**Comment by BinHawkins42** [comment]
ok I don't know if these have been a problem already or if they are new issues but there is a bug were the blocks will be off the 25cm on child entity grids making it so when you place a 1/4 block it tries to place it in between two blocks. 

![20210509224529_1](https://user-images.githubusercontent.com/36099433/117603814-93fc0d00-b119-11eb-9fa7-65d5429d790e.jpg)
![20210509224455_1](https://user-images.githubusercontent.com/36099433/117603817-9494a380-b119-11eb-8e4b-24f61d0aaf07.jpg)
![20210509224509_1](https://user-images.githubusercontent.com/36099433/117603819-952d3a00-b119-11eb-89bf-e673fe299443.jpg)

and then here is what it should look like.

![20210509224546_1](https://user-images.githubusercontent.com/36099433/117603837-a70edd00-b119-11eb-867a-73d909b6a33a.jpg)

then there appears to be a problem with the Gatling barrel for the laser. the model is not showing up and the weapon discharge is appearing about a meter away from where the barrel would be and a little above. 

![20210509224705_1](https://user-images.githubusercontent.com/36099433/117604089-1dabda80-b11a-11eb-8993-ab35ee3d9c9f.jpg)
![20210509225014_1](https://user-images.githubusercontent.com/36099433/117604111-28666f80-b11a-11eb-8282-a0f97f644806.jpg)

happy hunting. Keep up the good work. love the game so far. This update has potential to shake the vary foundation of block building community.

**Comment by tsunamayo** [comment]
@Kaiser-Indrasil yeah the reflection shape is a limitation that could addressed in the future, it is not the priority atm and would require some optimization first as the shader pass would be more expensive... Thanks

**Comment by Sebateses** [comment]
Love the new block scaling system Tsuna and the new lighting looks awesome! Thanks for adding the tiles back in too! Keep up the good work!

**Comment by Uncle-Ulty** [comment]
I've reported that reflection problem on the last 2 rounds. It seems that only the light tube has the reflection properly implemented. Maybe it's affected by the object orientation.

About the block's scaling and grid, wouldn't be easier to be able to choose which grid you wanna use? For instance, "T" for size and "Y" to select the grid.

And a suggestion for the future. Each different weapon size should have its own shooting sound effect. Bass for larger weapons and treble for smaller ones...

**Comment by ProPeach** [comment]
- This is likely a much longer standing bug rather than one introduced recently, but the knockback a ship receives from getting shot is in the opposite direction - Shoot a ship, and it starts drifting towards you, not away from you

Still plenty of outstanding bugs from the previous threads, but I think we can see the light at the end of the tunnel

**Comment by ExodistSKY1** [comment]
That would be fantastic!  Would you use this tech for the Custom Torus's (Habitat exteriors) you had mentioned earlier?

**Comment by tsunamayo** [comment]
- piston issue fixed.

**Comment by ExodistSKY1** [comment]
1/4th scale pistons are still having closing issues once they are placed and activated

**Comment by ExodistSKY1** [comment]
![20210510202543_1](https://user-images.githubusercontent.com/45324484/117744432-49889800-b1ce-11eb-944c-1a515ec69b02.jpg)

[Battlesuit core share to Tsuna for weird 18th building on hinge.zip](https://github.com/tsunamayo/Starship-EVO/files/6455865/Battlesuit.core.share.to.Tsuna.for.weird.18th.building.on.hinge.zip)

Hinge is still having collision/ not-placing issues 

also

the length of turn it takes to stretch the 1/8th bricks I feel is far too much

**Comment by RockefellerDoctor** [comment]
Am still experiencing the unusually high GPU usage/ low fps when viewing the builder drone ship preview in the codex. I thought this issue was fixed ages ago?

**Comment by tsunamayo** [comment]
@RockefellerDoctor one which kind of blueprint (for the fps issue)? Could you send me it? Also a bp for the gravity unit could be useful. thx

**Comment by tsunamayo** [comment]
- paint VFX fixed.

**Comment by tsunamayo** [comment]
- MMB menu paint color fixed @Kaiser-Indrasil ;)

**Comment by tsunamayo** [comment]
- Gravity Unit fixed.

**Comment by tsunamayo** [comment]
@BinHawkins42 I cant reproduce the children entity offset issue. What kind of mechanism you used? Can you send me the blueprint?

**Comment by tsunamayo** [comment]
- Washed out colors for glowing bricks fixed.

**Comment by tsunamayo** [comment]
- Collisions on pistons fixed
- Collisions on hinge fixed.
